_posts/2021-04-08-pulp-3.12-is-generally-available.md

```diff
@@ -0,0 +1,119 @@
+
+In previous versions of Pulp, no one could see deprecation warnings for features. With this release, if the log level is set to `WARNING`, you'll now find deprecation warnings included in your logs.
+
+You can disable this by adjusting the log level of `pulpcore.deprecation`.
+
+## Updates to the Plugin API
+
+This release includes the following updates to the Plugin API.
+
+### New callback method for autopublishing [#7626](https://pulp.plan.io/issues/7626)
+
+A new callback method ``on_new_version()`` has been added to ``Repository``. This runs when a new repository version has been created. You can use this method for automatically publishing or distributing the new repository version after it has been created.
+
+### New logging options for ``DigestValidationError`` and ``SizeValidationError`` [#8357](https://pulp.plan.io/issues/8357)
+
+A `url` option has been added as a possible argument to ``DigestValidationError`` and ``SizeValidationError`` exceptions, so that you can better locate the url of the requested files.
+
+### A new Distribution MasterModel [#8384](https://pulp.plan.io/issues/8384)
+
+This release provides a new MasterModel `pulpcore.plugin.models.Distribution` that replaces the BaseDistribution `pulpcore.plugin.models.BaseDistribution` model.
+You will have to define a migration to move your data from `pulpcore.plugin.models.BaseDistribution` to `pulpcore.plugin.models.Distribution`. See the `pulp_file` migration 0009 as a reference example.[#8385](https://pulp.plan.io/issues/8385)
+
+This new model contains the `repository`, `repository_version`, and `publication` fields  instead of on the detail models as was done with `pulpcore.plugin.models.BaseDistribution`.
+
+
+* A new serializer `pulpcore.plugin.serializer.DistributionSerializer` has been added to use with `pulpcore.plugin.models.Distribution`.
+* The viewset `pulpcore.plugin.viewset.DistributionViewSet`  replaces the deprecated `pulpcore.plugin.viewset.BaseDistributionViewSet`.
+* The new `pulpcore.plugin.viewset.NewDistributionFilter` filter pairs with the `Distribution` model.
+
+### Object deprecations as part of this change
+
+|BaseDistribution object|Replaced by|
+| --- | --- |
+|`pulpcore.plugin.models.BaseDistribution` |`pulpcore.plugin.models.Distribution` |
+|`pulpcore.plugin.viewset.BaseDistributionViewSet`|`pulpcore.plugin.viewset.DistributionViewSet` |
+|`pulpcore.plugin.serializer.BaseDistributionSerializer`|`pulpcore.plugin.serializer.DistributionSerializer`|
+|`pulpcore.plugin.serializer.PublicationDistributionSerializer`|`pulpcore.plugin.serializer.DistributionSerializer` |
+|`pulpcore.plugin.serializer.RepositoryVersionDistributionSerializer` | Define the `publication` field directly on your detail distribution object.|
+|`pulpcore.plugin.viewset.DistributionFilter` |`pulpcore.plugin.viewset.NewDistributionFilter`|
+|`pulpcore.plugin.tasking.enqueue_with_reservation` |`pulpcore.plugin.tasking.dispatch` |
```

Comment:
This last deprecation is part of another change.
But i like the list being a table of deprecations for the entire release. So maybe adjust the headers instead.

Not sure how to add the "task arguments must be JSON serializable" deprecation into that table.

Reply:
Lol I will aim for that for lucky 3.13. I just removed this and wrote about it in the deprecations section. Thanks for catching this!